### PR TITLE
Dala 4766 display quality when only quality

### PR DESCRIPTION
**Pull request by emanueldfine**
# Pull Request \<Title>
`<Description here>`
## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github. 
- [x] A peer-review has been executed
  - [x] The code has been manually inspected by someone who did not implement the feature
  - [x] If this PR includes work on the frontend, at least one `@ts-nocheck` is removed. Additionally, there should not be any `@ts-nocheck` in files modified by this PR. If no `@ts-nocheck` are left: Celebrate :tada: :confetti_ball: type-safety and remove this entry. 
- [x] The PR actually implements what is described in the JIRA-Issue
- [x] At least one test exists testing the new feature
  - [x] If you have created new test files, make sure that they are included in a test container and actually run in the CI
- [x] At least 3 consecutive CI runs are successfully executed to ensure that there are no flaky tests.
- [x] Documentation is updated as required
- [ ] The automated deployment is updated if required
- [ ] If there was a database entity class added, there must also be a migration script for creating the corresponding database if flyway is already used by the service
- [ ] IF there are changes done to the framework data models or to a database entity class, the following steps were completed in order
  - [ ] A fresh clone of dataland.com is generated (see Wiki page on "OTC" for details)
  - [ ] The feature branch is deployed to clone with `Reset non-user related Docker Volumes & Re-populate` turned off
  - [ ] It's verified that the CD run is green
  - [ ] The new feature is manually used/tested/observed on the clone server. Testing of the new feature should (also) be done by a second, independent reviewer from the dev team
  - [ ] The feature branch is deployed to dev1 with `Reset non-user related Docker Volumes & Re-populate` turned on, and it's verified that the CD run is green  
- [x] ELSE, the new version is deployed to the dev server "dev1" using this branch
  - [x] Run with setting `Reset non-user related Docker Volumes & Re-populate` turned on 
  - [x] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
  - [x] It's verified that the CD run is green
  - [x] The new feature is manually used/tested/observed on dev server. Testing of the new feature should (also) be done by a second, independent reviewer from the dev team

### CODE REVIEW

File: dataland-frontend/src/components/resources/dataTable/conversion/DataPoints.ts [hunk 1]
```diff
@@ -139,3 +133,37 @@ function doesAnyDataPointPropertyExist(dataPointProperties: DatapointProperties
...
+  fieldLabel: string,
+  datapointProperties: DatapointProperties | undefined | null
+): MLDTDisplayComponentTypes[MLDTDisplayComponentName.DataPointWrapperDisplayComponent] {
+  let innerContent: AvailableMLDTDisplayObjectTypes;
+  const isOnlyQualityProvided =
+    datapointProperties?.dataSource == undefined &&
+    datapointProperties?.comment == undefined &&
+    inputValue.displayValue == '';
+
+  if (isOnlyQualityProvided) {
```
Comment (dennis-dfine) [inline dataland-frontend/src/components/resources/dataTable/conversion/DataPoints.ts:156]: If I understand the code correctly, neither quality has to be provided. Consider
```suggestion
  if (isAtMostQualityProvided) {
```

File: dataland-frontend/src/components/resources/dataTable/conversion/DataPoints.ts [hunk 2]
```diff
@@ -139,3 +133,37 @@ function doesAnyDataPointPropertyExist(dataPointProperties: DatapointProperties
...
+    innerContent = formatStringForDatatable(datapointProperties?.quality);
+  } else {
+    innerContent = inputValue.displayValue == '' ? formatStringForDatatable(ONLY_AUXILIARY_DATA_PROVIDED) : inputValue;
+  }
+
+  return {
+    innerContents: innerContent,
+    quality: isOnlyQualityProvided ? undefined : datapointProperties?.quality ?? undefined,
```
Comment (dennis-dfine) [inline dataland-frontend/src/components/resources/dataTable/conversion/DataPoints.ts:164]: I don't understand. If quality is provided (`isOnlyQualityProvided == true`) you set quality to `undefined`? And if it is false, i.e. if eg. displayValue is not empty, you set it to `datapointProperties.quality` unless it is null? This means you set quality to undefined even if datapointProperties.quality exists and isOnlyQualityProvided is true, right? Why?

File: dataland-frontend/tests/e2e/fixtures/frameworks/sfdr/SfdrPreparedFixtures.ts [hunk 3]
```diff
@@ -198,3 +198,24 @@ function generateFixtureWithIncompleteReferencedReport(input: FixtureData<SfdrDa
...
 }
+
+/**
+ * Generates an SFDR dataset with three data points that only contain a value, a quality and a comment.
+ * @param input Fixture data to be manipulated
+ * @returns the dataset
+ */
+function generateFixtureWithDifferentExtendedDatapointCases(input: FixtureData<SfdrData>): FixtureData<SfdrData> {
+  input.companyInformation.companyName = 'TestForDataPointDisplayLogic';
+  if (input.t.environmental?.greenhouseGasEmissions?.scope1GhgEmissionsInTonnes) {
```
Comment (dennis-dfine) [inline dataland-frontend/tests/e2e/fixtures/frameworks/sfdr/SfdrPreparedFixtures.ts:209]: general info for me: why are these if-cases necessary? I mean, if these if-statements returned false, you wouldn't create an appropriate fixture right?

File: dataland-frontend/src/components/resources/dataTable/conversion/DataPoints.ts (hunk 2 above)
Comment (emanueldfine) [reply dataland-frontend/src/components/resources/dataTable/conversion/DataPoints.ts:165]: Ich trickse die Anzeigekomponente aus:

Wenn NUR quality da ist, dann möchte ich, dass die Anzeigekomponente das wie einen "Wert" anzeigt.
Daher verpacke ich in diesem Fall den String, der in `dataProperties.quality` steckt in den `innerContents`-Slot, und nehme ihn aus dem `quality` slot raus.

=> Das führt dazu, dass der Quality-String literally im Frontend als "Wert" auftraucht.


Hatte vorher einige andere Wege ausprobiert, das zu erreichen, die aber in Katastrophen geendet sind.
Grund: Der Anzeige-Code von Datapoints ist unfassbar verschachtelt und kleinste Änderungen sprengen Dinge an x Stellen. Daher habe ich den belassen wie er ist, und stattdessen an dieser Stelle hier direkt an der "Quelle" meine Änderungen vorgenommen.

File: dataland-frontend/tests/e2e/fixtures/frameworks/sfdr/SfdrPreparedFixtures.ts (hunk 3 above)
Comment (emanueldfine) [reply dataland-frontend/tests/e2e/fixtures/frameworks/sfdr/SfdrPreparedFixtures.ts:209]: Ja ich verstehe dass das komisch wirkt.
Grund ist, dass typescript dir nur erlaubt, Felder zu setten, wenn auch sichergestellt ist, dass das Objekt existiert.
Wenn ich das if wegnehme, wird typescript meckern.

Es gibt jetzt einige Alternativmöglichkeiten (Objekt literally bauen, Code schreiben à-la "wenns-Objekt-nicht-gibt-erstelle-es" etc.).
Aber die kommen alle nicht mit so wenigen Zeilen aus wie das hier 😄 

Da diese maniuplator-Funktion auf ein "volles" datset angewendet wird, sollte es auch immer klappen.

File: dataland-frontend/src/components/resources/dataTable/conversion/DataPoints.ts (hunk 2 above)
Comment (emanueldfine) [reply dataland-frontend/src/components/resources/dataTable/conversion/DataPoints.ts:166]: Wie besprochen: Habe die description dieser Funktion jetzt so angepasst, dass man dieses "Austricksen" hier als Code-Leser versteht.

**Pull request merged**